### PR TITLE
crimson/os/seastore: fix ordered updates to JournalSegmentManager::committed_to

### PR DESCRIPTION
**Pull request by cyx1231st**
Journal segment should not update committed_to during rolling as there
might be still pending writes from the previous segment.

A side-effect here is that committed_to now needs to include
segment_seq_t to point to a previous segment.

Signed-off-by: Yingxin Cheng <yingxin.cheng@intel.com>

---

See discussions in https://github.com/ceph/ceph/pull/43617#discussion_r738135557

This PR depends on the refactoring work in https://github.com/ceph/ceph/pull/43617 and starts from the last commit.

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by athanatos** [comment]
I think my original intention was for the close() in roll() to be a barrier -- I think it actually would be on a zns device.  block.cc doesn't actually enforce that though.

In any event, doing it this way enables us to begin writing to the next segment before the prior segment flushes and closes.

**Comment by athanatos** [comment]
LGTM once the other PR merges

**Comment by cyx1231st** [comment]
jenkins test make check

**Review by athanatos — APPROVED** [review]

**Review by xxhdx1985126 — APPROVED** [review]

**Pull request merged**
